app/controllers/application_controller.rb

```diff
@@ -20,11 +20,7 @@ def redirect_after_login
     elsif session[:return_to]
       redirect_to session[:return_to]
     else
-      @order = Order.create(user_id: current_user.id, status: "ordered")
-      @order.create_order_items(@cart)
-      @order.update_attributes(total_price: @order.order_total)
-      @cart.clear
-      redirect_to order_path(@order)
+      redirect_to new_order_path
```

Comment:
Created an order review page before submit which was able to get rid of this hack of creating an order here.